app/controllers/comments_controller.rb

```diff
@@ -1,10 +1,11 @@
 class CommentsController < ApplicationController
 
   def create
-    @question = Question.find(params[:comment][:question])
-    @comment = @question.comments.new(comment_params)
+    @presenter = Presenter.new
+    @question = Question.find(comment_params[:question])
+    @answer = Answer.populate_answer(comment_params)
+    @comment = Comment.populate_comment(comment_params, @question, @answer)
     @comment.user = current_user
```

Comment:
oof, this is one ugly controller method. I agree that it will bit a PITA to fix though.